### ISSUE TITLE
Proxy object fails to be a key in WeakMap

### ISSUE THREAD

**Issue by gullerya** [Microsoft/ChakraCore#2419]
Way to reproduce:
``` javascript
let o = {};
let p = new Proxy(o, {});
let m = new WeakMap();
m.set(p, o);
m.get(p);    // undefined, it works just fine in other browsers and there is no disclaimer in a spec that is shouldn't
```

**Comment by akroshg** [comment]
@gullerya , could you verify if this is fixed with this PR #2019 ?

**Comment by gullerya** [comment]
I'm not familiar with interim Chakra implementation, so it's hard to me to judge if the changes are really resolving the issue, but from the changes and the tests along the changes are indeed looks like it's addressing the issue.
I'm experiencing this problem on Microsoft Edge 38.14393.0.0 (Microsoft EdgeHTML 14.14393), so it's not shipped yet to this release?
I not - when the fix is expected to be delivered (what are the delivery cycles of Edge in general)?

**Comment by akroshg** [comment]
The fix is in for chakracore release v1.4 and this fix will certainly be available on the next Edge release. However I am not aware of the timelines and dates for next Edge release. You could try the latest Windows 10 insider preview.

**Comment by dilijev** [comment]
Verified the fix in the latest ChakraCore with the following modified test case:

```js
let o = { a: 42 };
let p = new Proxy(o, {});
let m = new WeakMap();
m.set(p, o);
console.log(JSON.stringify(m.get(p)));`
```

Output: `{"a":42}`
